includes/admin/home/tasks/class-sensei-home-tasks-provider.php

```diff
@@ -88,6 +90,49 @@ private function map_task( Sensei_Home_Task $task ): array {
 		];
 	}
 
+	/**
+	 * Return the site information needed for the task list component.
+	 *
+	 * @return array The site info, including title and image (which is the custom logo) URl.
+	 */
+	private function get_site() {
+		$custom_logo_id = get_theme_mod( 'custom_logo' );
+		$image          = wp_get_attachment_image_src( $custom_logo_id, 'full' );
+		return [
+			'title' => get_bloginfo( 'name' ),
+			'image' => $image ? $image[0] : null,
+		];
+	}
+
+	/**
+	 * Return the course information needed for the task list component, including title and image (which is the
+	 * featured image) for that course. Please note that the data for the demo course is never returned by this method.
+	 *
+	 * @return array|null The course information, including title and image URL.
+	 */
+	private function get_course() {
+		global $wpdb;
+		$cache_key   = 'home/metadata/tasks/course';
+		$cache_group = 'sensei/temporary';
+		$result      = wp_cache_get( $cache_key, $cache_group );
+		if ( false === $result ) {
+			$prefix = $wpdb->esc_like( Sensei_Data_Port_Manager::SAMPLE_COURSE_SLUG );
+			// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery -- Safe-ish and rare query.
+			$post_id = $wpdb->get_var( $wpdb->prepare( "SELECT ID FROM {$wpdb->posts} WHERE post_type='course' AND post_status IN ('publish', 'draft') AND post_name NOT LIKE %s ORDER BY post_status='published' DESC, ID ASC LIMIT 1", "{$prefix}%" ) );
+			if ( null === $post_id ) {
+				$result = null;
+			} else {
+				$image  = get_the_post_thumbnail_url( $post_id, 'full' );
+				$result = [
+					'title' => get_the_title( $post_id ),
+					'image' => $image ? $image : null,
```

Comment:
I just realized that we also need the permalink to the course. We'll need it to the share links in the ready state.

Reply:
Hey, thanks for the review! I added the permalink field in 9733db5ff1e9a20d9fc85c74bf886839972cffcc  :)